### ISSUE TITLE
Session info endpoint returns undocumented check keys

### ISSUE THREAD

**Issue by robn** [hashicorp/consul#7341]
## Overview of the Issue

The `/v1/session/info` endpoint returns `ServiceChecks` and `NodeChecks` keys, and does not return `Checks` [as documented](https://www.consul.io/api/session.html).

Change appears to have been introduced in #6909.

#### Reproduction Steps

Start a server:

```
$ perl -MTest::Consul -E '$tc = Test::Consul->start(port => 8500); sleep 86400'
==> Starting Consul agent...
           Version: 'v1.7.1'
           Node ID: '99846629-02db-889f-c538-4657ac5593b4'
         Node name: 'tc_node1'
        Datacenter: 'tc_dc1' (Segment: '<all>')
            Server: true (Bootstrap: false)
       Client Addr: [127.0.0.1] (HTTP: 8500, HTTPS: -1, gRPC: -1, DNS: -1)
      Cluster Addr: 127.0.0.1 (LAN: 49153, WAN: 49154)
           Encrypt: Gossip: false, TLS-Outgoing: false, TLS-Incoming: false, Auto-Encrypt-TLS: false
```

Create a session:

```
$ curl -X PUT http://127.0.0.1:8500/v1/session/create
{
    "ID": "1ef29c5f-719b-7b59-e486-d6d1c797468e"
}
```

Get session info:

```
$ curl http://127.0.0.1:8500/v1/session/info/1ef29c5f-719b-7b59-e486-d6d1c797468e
[
    {
        "ID": "1ef29c5f-719b-7b59-e486-d6d1c797468e",
        "Name": "",
        "Node": "tc_node1",
        "LockDelay": 15000000000,
        "Behavior": "release",
        "TTL": "",
        "NodeChecks": [
            "serfHealth"
        ],
        "ServiceChecks": null,
        "CreateIndex": 16,
        "ModifyIndex": 16
    }
]
```

Note the keys do not match the documentation.

### Consul info for both Client and Server

<details>
  <summary>Client info</summary>

```
agent:
	check_monitors = 0
	check_ttls = 0
	checks = 0
	services = 0
build:
	prerelease = 
	revision = 2cf0a3c8
	version = 1.7.1
consul:
	acl = disabled
	bootstrap = false
	known_datacenters = 1
	leader = true
	leader_addr = 127.0.0.1:49155
	server = true
raft:
	applied_index = 93
	commit_index = 93
	fsm_pending = 0
	last_contact = 0
	last_log_index = 93
	last_log_term = 2
	last_snapshot_index = 0
	last_snapshot_term = 0
	latest_configuration = [{Suffrage:Voter ID:99846629-02db-889f-c538-4657ac5593b4 Address:127.0.0.1:49155}]
	latest_configuration_index = 0
	num_peers = 0
	protocol_version = 3
	protocol_version_max = 3
	protocol_version_min = 0
	snapshot_version_max = 1
	snapshot_version_min = 0
	state = Leader
	term = 2
runtime:
	arch = amd64
	cpu_count = 8
	goroutines = 74
	max_procs = 8
	os = linux
	version = go1.13.7
serf_lan:
	coordinate_resets = 0
	encrypted = false
	event_queue = 1
	event_time = 2
	failed = 0
	health_score = 0
	intent_queue = 0
	left = 0
	member_time = 1
	members = 1
	query_queue = 0
	query_time = 1
serf_wan:
	coordinate_resets = 0
	encrypted = false
	event_queue = 0
	event_time = 1
	failed = 0
	health_score = 0
	intent_queue = 0
	left = 0
	member_time = 1
	members = 1
	query_queue = 0
	query_time = 1
```

</details>

<details>
  <summary>Server info</summary>

```
$  consul info
agent:
	check_monitors = 0
	check_ttls = 0
	checks = 0
	services = 0
build:
	prerelease = 
	revision = 2cf0a3c8
	version = 1.7.1
consul:
	acl = disabled
	bootstrap = false
	known_datacenters = 1
	leader = true
	leader_addr = 127.0.0.1:49155
	server = true
raft:
	applied_index = 93
	commit_index = 93
	fsm_pending = 0
	last_contact = 0
	last_log_index = 93
	last_log_term = 2
	last_snapshot_index = 0
	last_snapshot_term = 0
	latest_configuration = [{Suffrage:Voter ID:99846629-02db-889f-c538-4657ac5593b4 Address:127.0.0.1:49155}]
	latest_configuration_index = 0
	num_peers = 0
	protocol_version = 3
	protocol_version_max = 3
	protocol_version_min = 0
	snapshot_version_max = 1
	snapshot_version_min = 0
	state = Leader
	term = 2
runtime:
	arch = amd64
	cpu_count = 8
	goroutines = 74
	max_procs = 8
	os = linux
	version = go1.13.7
serf_lan:
	coordinate_resets = 0
	encrypted = false
	event_queue = 1
	event_time = 2
	failed = 0
	health_score = 0
	intent_queue = 0
	left = 0
	member_time = 1
	members = 1
	query_queue = 0
	query_time = 1
serf_wan:
	coordinate_resets = 0
	encrypted = false
	event_queue = 0
	event_time = 1
	failed = 0
	health_score = 0
	intent_queue = 0
	left = 0
	member_time = 1
	members = 1
	query_queue = 0
	query_time = 1
```

</details>

### Operating system and Environment details

Linux.

**Comment by jsosulska** [comment]
Hi @robn , Thanks for this issue, and the PR. I'll have someone look at your PR to provide feedback. This may just need to be the documentation being updated based on the new checks' functionality.

**Comment by ahamlinman** [comment]
We ran into this behavior following an upgrade to v1.7.2. In particular the lack of a `Checks` key in the session info response broke one of our Consul-based leader election implementations.

It looks like `Checks` only disappears from the info response when the session was created with `Checks` set to the default value of `["serfHealth"]` (implicitly or explicitly). When the session is associated with additional checks they're returned in a `Checks` key when querying session info just as in previous Consul versions.

I'm _guessing_ the behavior is linked to [this snippet in `fixupEmptySessionChecks` of `agent/session_endpoint.go`](https://github.com/hashicorp/consul/blob/v1.7.2/agent/session_endpoint.go#L215-L219), which may be worth looking into in addition to any documentation changes. (I would assume the snippet has some desirable effect elsewhere, though I couldn't tell what that might be.)

```go
// If the checks field contains the default value, empty it out. Defer to what is in NodeChecks.
if len(session.Checks) == 1 && session.Checks[0] == structs.SerfCheckID {
	session.Checks = nil
	return
}
```

(For what it's worth, this is an edge case for us. The vast majority of sessions set up by our implementation were associated with additional checks and were unaffected. We've set up a workaround in our implementation for the case where `Checks` is missing.)
